File: LLM-Translations/GPT-4/avatar/Java/Go/codeforces_665_A.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	aTokens := readTokens(reader)
	a, _ := strconv.Atoi(aTokens[0])
	tA, _ := strconv.Atoi(aTokens[1])

	bTokens := readTokens(reader)
	b, _ := strconv.Atoi(bTokens[0])
	tB, _ := strconv.Atoi(bTokens[1])

	timing, _ := reader.ReadString('\n')
	timing = strings.TrimSpace(timing)
	hrs, _ := strconv.Atoi(timing[:2])
	mins, _ := strconv.Atoi(timing[3:])

	simDeparture := hrs*60 + mins
	simArrival := simDeparture + tA
	counter := 0

	for i := 300; i < 1440; i += b {
		busDeparture := i
		busArrival := i + tB

		if busDeparture >= simArrival || simDeparture >= busArrival {
			continue
		}
		counter++
	}

	fmt.Fprintln(writer, counter)
}

func readTokens(reader *bufio.Reader) []string {
	line, _ := reader.ReadString('\n')
	line = strings.TrimSpace(line)
	return strings.Split(line, " ")
}

// //End of Code

File: LLM-Translations/GPT-4/avatar/Java/Python/codeforces_665_A.py

from sys import stdin

def main():
    a, tA = map(int, stdin.readline().split())
    b, tB = map(int, stdin.readline().split())
    timing = stdin.readline().strip()
    hrs = int(timing[:2])
    mins = int(timing[3:])
    
    sim_departure = hrs * 60 + mins
    sim_arrival = sim_departure + tA
    
    counter = 0
    for i in range(300, 1440, b):
        bus_departure = i
        bus_arrival = i + tB
        if bus_departure >= sim_arrival or sim_departure >= bus_arrival:
            continue
        counter += 1
    
    print(counter)

if __name__ == "__main__":
    main()


